xcm/xcm-executor/src/traits/weight.rs

```diff
@@ -43,6 +43,18 @@ pub trait WeightBounds<Call> {
 	/// any internal effects. Inner XCM messages may be executed by:
 	/// - Order::BuyExecution
 	fn deep(message: &mut Xcm<Call>) -> Result<Weight, ()>;
+
+	/// Return the total weight for executing `message`.
+	fn weight(message: &mut Xcm<Call>) -> Result<Weight, ()> {
+		Self::shallow(message)?.checked_add(Self::deep(message)?).ok_or(())
+	}
+}
+
+/// A means of getting approximate weight consumption for a given destination message executor and a
+/// message.
+pub trait UniversalWeigher {
```

Comment:
I read this from my phone so not 100% sure, but I can't find this trait being used.

Reply:
Yeah, it's not referenced yet, but I expect it'll be useful in the future. Can remove if it offends.